jenkins_pipelines/scripts/SUSEManager_cleaner/suse_manager_cleaner_program/SUSEManagerCleaner.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python3
+import argparse
+import logging
+import sys
+from suse_manager_api import ResourceManager
+from suse_manager_ssh import SSHClientManager
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Define the available modes
+MODES = [
+    'delete_users', 'delete_activation_keys', 'delete_config_projects',
+    'delete_software_channels', 'delete_systems', 'delete_repositories',
+    'full_cleanup', 'delete_salt_keys', 'delete_known_hosts',
+    'update_custom_repositories', 'delete_distributions'
+]
+
+def main():
+    parser = argparse.ArgumentParser(description="Manage SUSE Manager API actions.")
+    parser.add_argument("--url", required=True, help="The URL of the SUSE Manager XML-RPC API.")
+    parser.add_argument("--mode", required=True, choices=MODES, help="The mode of operation.")
+    parser.add_argument("--default-resources-to-delete", type=str, nargs='*',
+                        choices=['proxy', 'monitoring-server', 'retail'],
+                        default=[], help='List of default modules to force deletion')
+    parser.add_argument("--product_version", required=True, type=str, choices=['5.0', '4.3', 'uyuni'])
+
+    args = parser.parse_args()
+    manager_url = args.url
+
+    # API part
+    if args.mode in ["delete_users", "delete_activation_keys", "delete_config_projects",
+                     "delete_software_channels", "delete_systems", "delete_repositories",
+                     "full_cleanup", "delete_salt_keys"]:
+        resource_manager = ResourceManager(manager_url, args.default_resources_to_delete, args.product_version)
+        resource_manager.get_session_key()
+        mode_actions = {
+            "delete_users": resource_manager.delete_users,
+            "delete_activation_keys": resource_manager.delete_activation_keys,
+            "delete_config_projects": resource_manager.delete_config_projects,
+            "delete_software_channels": resource_manager.delete_software_channels,
+            "delete_systems": resource_manager.delete_systems,
+            "delete_repositories": resource_manager.delete_channel_repos,
+            "delete_salt_keys": resource_manager.delete_salt_keys,
+            "full_cleanup": resource_manager.run,
+        }
+        try:
+            action = mode_actions.get(args.mode)
+            if action:
+                action()
+            else:
+                logger.error(f"Mode '{args.mode}' is not recognized.")
+        finally:
+            resource_manager.logout_session()
+
+    # Server commands part
+    else:
+        ssh_manager = SSHClientManager(url=manager_url)
+        try:
+            if args.mode == "delete_known_hosts":
+                if args.product_version == "4.3":
+                    ssh_manager.run_command("rm /var/lib/salt/.ssh/known_hosts")
```

Comment:
I would follow the same mechanism to trigger actions for all the cases, so as we do here:

```
action = mode_actions.get(args.mode)
if action:
    action()
```

Then have an instance of SSHClientManager inside ResourceManager (or passing it on the constructor) and use it in the methods that need it.

Reply:
Done. Full ssh module change and actions in main code.